target/linux/ar71xx/image/tp-link.mk

```diff
@@ -913,3 +913,16 @@ define Device/tl-wa7210n-v2
     CONSOLE := ttyATH0,115200
 endef
 TARGET_DEVICES += tl-wa7210n-v2
+
+define Device/wpa8630
+    $(Device/tplink-8mlzma)
+    DEVICE_TITLE := TP-LINK WPA8630
+    DEVICE_PACKAGES := kmod-ath10k ath10k-firmware-qca988x
+    BOARDNAME := WPA8630
+    DEVICE_PROFILE := WPA8630
+    TPLINK_HWID := 0x86300001
+    MTDPARTS = spi0.0:64k(u-boot)ro,1280k(kernel),6720k(rootfs),64k(mib0)ro,64k(ART)ro,8000k@0x10000(firmware)
+    IMAGE/sysupgrade.bin := append-rootfs | mktplinkfw sysupgrade -a 0x10000
+    IMAGE/factory.bin := append-rootfs | mktplinkfw factory -a 0x10000
```

Comment:
I'm a bit confused. You wrote:

>Currently orginal firmware can be upgraded to LEDE using serial adapter, because tp-link used upgraded header v1 and mktplinkfw not support this.

but you have added a factory.bin here and you are using mktplinkfw. Can you please elaborate on this.